### PR TITLE
Major example overhaul and refactor

### PR DESCRIPTION
**Pull request by adriangl**
The PR includes the modifications in the architecture that we've been doing in the latest months, apart from other misc changes. Here's the changelog:
* Change package structure:
    * `network` contains network classes, such as network models or Retrofit APIs
    * `domain` contains classes related to business logic: Stores, Controllers, Actions or application models.
    * `ui` contains classes related to view logic, like `ViewModel`s, `Fragment`s or `Activity` classes.
    * Inside those packages we split sub-packages by functionality
* Change `AndroidViewModel`s with `ViewModel`s: this eases mocking of said `ViewModel`s in testing classes by removing the `Application` dependency with the Android framework.
* Add `@OpenForTesting` annotation so we can mock `ViewModel`s in Android Tests by opening the clases using `kotlin-allopen` plugin. For reference: https://proandroiddev.com/mocking-androidtest-in-kotlin-51f0a603d500
* Add very basic `ViewData` unit tests: remember that the idea of having `ViewData`s is that we can test the Store-to-View mapping logic with unit tests, and UI integration by mocking the `ViewModel` and simulating `LiveData` emissions in Espresso tests.
* Match package names and project name with the repo
* Update README references

### CODE REVIEW

**Review by finxo — APPROVED** [review]

File: app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt [hunk 1]
```diff
@@ -40,4 +47,20 @@ class MovesStore(private val movesController: MovesController) : Store<MovesStat
             )
         )
     }
+}
+
+object MovesModule {
+    fun create() = Kodein.Module("movesStore") {
```
Comment (yamidragut) [inline app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt]:
```suggestion
    fun create() = Kodein.Module("movesModule") {
```

File: app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt [hunk 2]
```diff
@@ -40,4 +47,20 @@ class MovesStore(private val movesController: MovesController) : Store<MovesStat
...
+        bindStore { MovesStore(instance()) }
+        bind<MovesController>() with singleton {
+            MovesControllerImpl(
+                instance(),
+                instance()
```
Comment (yamidragut) [inline app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt]: I would put it in the same line, there is no intereset in this, it is always going to be just `instance()`
`MovesControllerImpl(instance(), instance())`

Same in the other modules in other stores

File: app/src/main/kotlin/com/bq/arch_example/ui/pokemonlist/PokemonListActivity.kt [hunk 3]
```diff
@@ -47,10 +47,10 @@ class PokemonListActivity : AppCompatActivity(), KodeinAware {
 
         pokemonListViewModel.setup()
 
-        pokemonListViewModel.getPokemonListLiveData().observe(this) { viewData ->
-            toggleViewsVisibility(viewData.pokemonListRes, list_content, list_loading, error, View.GONE)
+        pokemonListViewModel.getPokemonListLiveData().observe(this) { resource ->
```
Comment (yamidragut) [inline app/src/main/kotlin/com/bq/arch_example/ui/pokemonlist/PokemonListActivity.kt]: Maybe the extension for the `observeResource` should be added to the template

Comment (adriangl) [reply app/src/main/kotlin/com/bq/arch_example/ui/pokemonlist/PokemonListActivity.kt]: 70e5a3c

File: app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt (hunk 2 above)
Comment (adriangl) [reply app/src/main/kotlin/com/bq/arch_example/domain/moves/MovesStore.kt]: 70e5a3c

File: app/src/main/kotlin/com/bq/arch_example/utils/extensions/LiveDataExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import mini.Resource
+
+
+/**
+ * Observes a given [LiveData].
+ */
+inline fun <T> LiveData<T>.observe(owner: LifecycleOwner, crossinline cb: (T) -> Unit) = observe(owner, Observer { cb(it) })
+
+/**
+ * Observes a given the [LiveData] only once.
```
Comment (yamidragut) [inline app/src/main/kotlin/com/bq/arch_example/utils/extensions/LiveDataExtensions.kt]:
```suggestion
 * Observes a given [LiveData] only once.
```

**Review by yamidragut — APPROVED** [review]

**Pull request merged**